docs/Submodules.md

```diff
@@ -0,0 +1,53 @@
+Submodules
+==========
+
+This repository composes ASP.NET Core repos from a collection of many submodules.
+Working with submodules in git introduces some quirks into an otherwise workflow.
+Here are some tips for working with submodules.
+
+## Cloning
+
+By default, submodules will not be present. Use `--recursive` to clone all submodules.
+
+    git clone https://github.com/aspnet/Universe.git --recursive
+
+If you have already cloned, run this to initialize all submodules.
+
+    git submodule update --init
+
+## Pulling updates
+
+When you execute `git pull`, submodules do not automatically snap to the version
+used in new commits of the master repo. Update all modules to match the master repo
+commit by running this command.
+
+    git submodule update
+
+## Executing a command on each submodule
+
+    git submodule foreach '<command>'
+
+For example, to clean and reset each submodule:
+
+    git submodule foreach 'git reset --hard; git clean -xfd'
+
+## Updating submodules
+
+Updating all submodules to newer versions can be done like this.
```

Comment:
How is the newer version determined in this case? Does it just take the latest of the branch for the submodule?

Reply:
Yup. it uses the branch as configured in .gitmodules